app/src/main/java/org/schabi/newpipe/player/helper/AudioReactor.java

```diff
@@ -149,11 +151,21 @@ public void onAnimationEnd(final Animator animation) {
 
     @Override
     public void onAudioSessionId(final EventTime eventTime, final int audioSessionId) {
+        notifyAudioSessionUpdate(true, audioSessionId);
+    }
+
+    @Override
+    public void onAudioDisabled(final EventTime eventTime, final DecoderCounters counters) {
```

Comment:
Isn't there also a `onAudioEnabled`?

Reply:
there is, but it doesn't matter. When a new audioSessionId is generated, we have the method "onAudioSessionId" or "onAudioSessionIdChanged" in newer ExoPlayer versions. 

The only two reasons the audioSessionId gets reset is in onAudioDisabled and on destroy of the Player. 

So all cases are already handled with  onAudioSessionId, dispose and onAudioDisabled.

Btw. that the audiosessionId gets reset in onAudioDisabled is a bug in ExoPlayer. This commit here states what they wanted to do since 2.13.0: https://github.com/google/ExoPlayer/commit/a95b2ebb1d3125a4899c0dfe4a830077271ebca0

in 2.15.0 they found their mistake as in onAudioDisabled the audioSessionId doesn't get reset.

So we need to remember to remove "onAudioDisabled" when we update to 2.15. or later.

Reply:
oh no, they fixed it already in a newer version. So onAudioDisabled is not required actually.

https://github.com/google/ExoPlayer/commit/f52e4bed3d9bcde9fb380855c54c437b22b98b6f#diff-5078a6f4b57d380b958c26c0850edd29043d7a10f48b17bb172e55a840a52bb8

Reply:
wait: at this state of the repository it IS required. We need to remove it once #7005 is merged.

Reply:
made some edits to my comments so that they are clearer

Reply:
Ok, I see